packages/libraries/linkable-specs/src/__tests__/index.spec.ts

```diff
@@ -0,0 +1,77 @@
+import { DocumentNode, parse, StringValueNode, visit } from 'graphql';
+import { detectLinkedImplementations, LinkableSpec } from '../index';
+
+test('LinkableSpec and detectLinkedImplementations can be used to easily implement linked schema functionality', () => {
+  const sdl = `
+        directive @meta(name: String!, content: String!) on SCHEMA | FIELD
+        directive @metadata__example(eg: String!) on FIELD
+        extend schema
+          @link(url: "https://specs.apollo.dev/federation/v2.3")
+          @link(url: "https://specs.graphql-hive.com/metadata/v0.1", import: ["@meta"])
+
+        type Query {
+          ping: String @meta(name: "owner", content: "hive-console-team")
+          pong: String @metadata__example(eg: "1...2...3... Pong")
+        }
+    `;
+
+  const metaSpec = new LinkableSpec('https://specs.graphql-hive.com/metadata', {
+    // The return value could be used to map sdl, collect information, or create a graphql yoga plugin.
+    // In this test, it's used to collect metadata information from the schema.
+    'v0.1': resolveImportName => (typeDefs: DocumentNode) => {
+      const collectedMeta: Record<string, Record<string, string>> = {};
+      const metaName = resolveImportName('@meta');
+      const exampleName = resolveImportName('@example');
+      visit(typeDefs, {
+        FieldDefinition: node => {
+          let metaData: Record<string, string> = {};
+          const fieldName = node.name.value;
+          const meta = node.directives?.find(d => d.name.value === metaName);
+          if (meta) {
+            metaData['name'] =
+              (
+                meta.arguments?.find(a => a.name.value === 'name')?.value as
+                  | StringValueNode
+                  | undefined
+              )?.value ?? '??';
+            metaData['content'] =
+              (
+                meta.arguments?.find(a => a.name.value === 'content')?.value as
+                  | StringValueNode
+                  | undefined
+              )?.value ?? '??';
+          }
+
+          const example = node.directives?.find(d => d.name.value === exampleName);
+          if (example) {
+            metaData['eg'] =
+              (
+                example.arguments?.find(a => a.name.value === 'eg')?.value as
+                  | StringValueNode
+                  | undefined
+              )?.value ?? '??';
+          }
+          if (Object.keys(metaData).length) {
+            collectedMeta[fieldName] ??= {};
+            collectedMeta[fieldName] = Object.assign(collectedMeta[fieldName], metaData);
+          }
+          return;
+        },
+      });
+      // collect metadata
+      return `running on v0.1.\nFound metadata: ${JSON.stringify(collectedMeta)}}`;
+    },
+    'v0.2': _resolveImportName => (_typeDefs: DocumentNode) => {
+      // collect metadata
+      return `running on v0.2...`;
+    },
+  });
+  const typeDefs = parse(sdl);
+  const linked = detectLinkedImplementations(typeDefs, [metaSpec]);
```

Comment:
In practice, what would typical use-cases be of an application for this library? Is it mostly for GraphQL platforms that are doing meta work on the schemas, or, would there be common use-cases for this in GraphQL applcations too, such as our own Hive GraphQL API?

(Maybe the answer to this question would make for good README content?)

Reply:
For power users / gateway developers who want to implement custom logic/directives using `@link`s

Reply:
Thanks, makes sense!